src/python/WMCore/WMSpec/WMWorkloadTools.py

```diff
@@ -215,7 +215,7 @@ def validateSiteLists(arguments):
         whiteList = [whiteList]
     if not isinstance(blackList, list):
         blackList = [blackList]
-    res = (set(whiteList) & set(blackList))
+    res = [s for s in list(set(whiteList) & set(blackList)) if s]
```

Comment:
Ah, thanks. I understand now. Lines 215 and 217 are the real problem. Could add "and blackList" and else blackList = []